keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlVerificationRepository.kt

```diff
@@ -162,6 +168,77 @@ class SqlVerificationRepository(
       }
     }
 
+  /**
+   * Query the repository for the states of multiple contexts.
+   *
+   * This call is semantically equivalent to
+   *    contexts.map { context -> this.getStates(context) }
+   *
+   * However, it's implemented as a single query for efficiency.
+   *
+   * @param contexts a list of verification contexts to query for state
+   * @return a list of maps of verification ids to states, in the same order as the contexts. If there are no
+   *         verification states associated with a context, the resulting map will be empty.
+   */
+  override fun getStatesBatch(contexts: List<VerificationContext>) : List<Map<String, VerificationState>> {
+    /**
+     * In-memory database table representation of the set of contexts we want to query for
+     *
+     * Columns:
+     *   ind - index that encodes the original list order, to ensure results are in same order
```

Comment:
Does it matter that it's in the same order?

Reply:
Yeah, the problem is that a `VerificationState` object doesn't carry information about which `VerificationContext` it's associated with, so if we just return a set of states, we don't know what they refer to.

We could modify `VerificationState` to have a reference to the context it's associated with.